src/main/java/math/Fraction.java

```diff
@@ -201,7 +201,15 @@ public Fraction reciprocal() {
      */
     public Fraction abs() {
         // TODO (BQYCKN)
-        return null;
+        int numerator = this.numerator;
+        int denominator = this.denominator;
+        if (numerator < 0){
+            numerator *= -1;
+        }
+        if (denominator < 0) {
```

Comment:
The `denominator` of a `Fraction` object is always positive, this is ensured by the constructor. Thus this check should be removed.